tests/funcalign/test_srm.py

```diff
@@ -103,3 +114,104 @@ def test_can_instantiate():
+    try:
+        model.transform(X)
+        assert True, "Success transforming before fitting the model!"
+    except NotFittedError:
+        print("Caught Exception number 1: transforming before fitting the model")
+
+    # Check that it does NOT run with 1 subject
+    try:
+        model.fit(X)
+        assert True, "Success running DetSRM with one subject!"
+    except ValueError:
+        print("Caught Exception number 1: running DetSRM with 1 subject")
+
+    for subject in range(1, subjects):
+        Q, R = np.linalg.qr(np.random.random((voxels, features)))
+        W.append(Q)
+        X.append(Q.dot(S) + 0.1*np.random.random((voxels, samples)))
+
+    # Check that runs with 2 subject
+    try:
+        model.fit(X)
+    except ValueError:
+        assert True, "Problem running DetSRM."
+
+    assert len(model.w_) == subjects, "Invalid computation of DetSRM! (wrong # subjects in W)"
+    for subject in range(subjects):
+        assert model.w_[subject].shape[0] == voxels, "Invalid computation of DetSRM! (wrong # voxels in W)"
+        assert model.w_[subject].shape[1] == features, "Invalid computation of DetSRM! (wrong # features in W)"
+    assert model.s_.shape[0] == features, "Invalid computation of DetSRM! (wrong # features in S)"
+    assert model.s_.shape[1] == samples, "Invalid computation of DetSRM! (wrong # samples in S)"
+
+    # Check that it does run to compute the shared response after the model computation
+    try:
+        new_s = model.transform(X)
+    except ValueError:
+        assert True, "Problem transforming new data with DetSRM."
+
+    assert len(new_s) == subjects, "Invalid computation of DetSRM! (wrong # subjects after transform)"
+    for subject in range(subjects):
+        assert new_s[subject].shape[0] == features, "Invalid computation of DetSRM! (wrong # features after transform)"
+        assert new_s[subject].shape[1] == samples, "Invalid computation of DetSRM! (wrong # samples after transform)"
+
+    # Check that it does NOT run with non-matching number of subjects
+    try:
+        model.transform(X[1])
+        assert True, "Success transforming with non-matching number of subjects"
+    except ValueError:
+        print("Caught Exception number 2: transforming with non-matching number of subjects")
+
+    # Check that it does not run without enough samples (TRs).
+    try:
+        model.set_params(features=(samples+1))
+        model.fit(X)
+        assert True, "Success running DetSRM with more features than samples!"
+    except ValueError as e:
+        print("Catched Exception number 3: not enough samples")
+
+    # Check that it does not run with different number of samples (TRs)
+    S2 = S[:,:-2]
+    X.append(Q.dot(S2))
+    try:
+        model.fit(X)
+        assert True, "Success running DetSRM with different number of samples!"
+    except ValueError:
+        print("Catched Exception number 2: different number of samples per subject")
+
+
```

Comment:
Can we add a test with the synthetic data to make sure the output of DetSRM is consistent. To be specific, a test to check that the output Ws are all orthonormal and W_i*S are close to input X within a threshold delta. This doesn't have to be in this PR but it's necessary to have one in case some future editing break the code.